src/wwwroot/startupscripts/startagent-mac.sh

```diff
@@ -2,6 +2,15 @@
 
 workspace_path=$1
 
+# Try to clean workspace 
+rm -r -f $workspace_path
+
+if [ -d "$workspace_path" ]; then
+	echo "Failed to clean workspace... requesting infra retry and reboot"
+	$HELIX_PYTHONPATH -c "from helix.workitemutil import request_infra_retry; request_infra_retry('Optional reason string')"
```

Comment:
If you're going to do this, you have to return; otherwise the code starting the agent below will invalidate the token for other machine names.